### PR TITLE
Feature/search bar state

### PR DESCRIPTION
**Pull request by sarahdherren**
## Description

As a user, I want the search bar to reflect the current search after clicking on a tag.
To implement this feature we created a new piece of state, an action, & a reducer and connected the value of the search input to update whenever the state is changed either by entering text into the input box directly and when a user searches by clicking on a tag.

[video](https://www.loom.com/share/08274d35046c43408b4797f8b5ed7678)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by stu-wd — APPROVED** [review]
this is really good work. great reinforcement to the user after clicking on a tag. you really dug into the ant-design docs and stackoverflow to figure what that `onSearch` property was all about. later in the week, maybe we can revisit to see if there is any way we can refactor this. overall, good work and good video description. the code is understandable and is properly explained.

File: src/components/common/Header.js [hunk 1]
```diff
@@ -39,18 +40,27 @@ function MainHeader(props) {
...
 
-  const { searchDocs } = props;
+  const { searchDocs, displayListView, displayThumbnail, searchTerm } = props;
   const {
     authService: { logout },
     authState,
   } = useOktaAuth();
   const { pathname } = useLocation();
 
+  useEffect(() => {
```
Comment (christina-yun) [inline src/components/common/Header.js:50]: Really like this that will make search more dynamic!

File: src/components/common/Header.js [hunk 2]
```diff
@@ -39,18 +40,27 @@ function MainHeader(props) {
...
 
-  const onSearch = value => {
+  const changeHandler = e => {
+    setQuery(e.target.value);
+  };
+
+  const onSearch = (value, e) => {
     if (!value) return alert('Search bar cannot be empty');
     searchDocs(value, authState);
+    e.target.value = '';
```
Comment (christina-yun) [inline src/components/common/Header.js:63]: More dynamic stuff -- awesome.

**Review by christina-yun — APPROVED** [review]
Overall this is really clean and all related to the search bar state. It's nice to see how you've put together all the code to make the search more dynamic which will be better UI for the user overall.

**Review by dominthebox — APPROVED** [review]
This is awesome, I was really looking forward to this feature working properly and this seems like an elegant solution. Nice job with the video demoing feature then showing how you all worked to implement it. It's exciting when all these little details are starting to come together.

**Review by Bobby-Tav — APPROVED** [review]
This code looks good. It is straightforward and concise which i really need to work on when it comes to my code. But overall it looks great and i i don't see any aberrations.

**Pull request merged**